1. Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [1]:
import numpy as np

In [3]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

In [4]:
cov = np.mean(zp * ks) - np.mean(zp) * np.mean(ks)
cov

9157.839999999997

In [5]:
np.cov(zp, ks, ddof=0)

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

In [8]:
sigma_zp = np.std(zp, ddof=0)
sigma_zp

59.115480206118605

In [9]:
sigma_ks = np.std(ks, ddof=0)
sigma_ks

174.55340157098058

In [11]:
r = cov / (sigma_zp * sigma_ks)
r

0.8874900920739158

In [12]:
np.corrcoef(zp, ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

2. Измерены значения IQ выборки студентов, обучающихся в местных технических вузах: 131, 125, 115, 122, 131, 115, 107, 99, 125, 111. Известно, что в генеральной совокупности IQ распределен нормально. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

Так как в генеральной совокупности IQ распределен нормально, а среднее квадратичное отклонение нам неизвестно, то для расчета доверительного интервала будем использовать критерий Стьюдента и несмещенную оценку дисперсии по имеющейся выборке.

$\overline{X} \pm t_{\alpha/2} \cdot \frac{\sigma_{н}}{\sqrt{n}}$

In [14]:
x = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
x_ = x.mean()
x_

118.1

In [15]:
sigma = x.std(ddof=1)
sigma

10.54566788359614

$\alpha=0.05$

$n=10$

Из таблицы критериев Стьюдента:

$t_{\alpha/2}=t_{0.025}=2.262$

Левая граница доверительного интервала:

In [16]:
x_ - 2.262 * sigma / 10**0.5

110.55660776308164

Правая граница доверительного интервала:

In [17]:
x_ + 2.262 * sigma / 10**0.5

125.64339223691834

Таким образом, искомый доверительный интервал:

$\bigl[ 110.557 ; 125.643 \bigr]$

3. Известно, что рост футболистов в сборной распределен нормально с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27, среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

Так как известно, что генеральная совокупность распределена нормально с $\sigma^2=25$, то для расчета доверительного интервала будем использовать $z$-критерий.

$\overline{X} \pm z_{\alpha/2} \cdot \frac{\sigma}{\sqrt{n}}$

$\overline{X} = 174.2$

$\sigma^2 = 25$, $\sigma = 5$

$n = 27$

$\alpha = 0.05$

Из $z$-таблицы $z_{0.025} = 1.96$

Таким образом, искомый доверительный интервал:

$\bigl[174.2-1.96\cdot \frac{5}{\sqrt{27}} ; 174.2+1.96\cdot \frac{5}{\sqrt{27}}\bigr]$

$\bigl[172.314 ; 176.086\bigr]$

In [18]:
174.2 - 1.96 * 5 / 27**0.5

172.31398912064722

In [19]:
174.2 + 1.96 * 5 / 27**0.5

176.08601087935276